src/current/advisories/a99561.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Technical Advisory 99561
+advisory: A-99561
+summary: When adding a new column using the new declarative schema changer, CockroachDB retrieves a unique secondary index mistakenly, assuming it to be a primary index.
+toc: true
+affected_versions: v22.2.x, v23.1.0-v23.1.5
+advisory_date: 2023-07-07
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+In CockroachDB v20.2 and v21.1 (the first two versions to support [`ALTER PRIMARY KEY`](../v21.1/alter-primary-key.html)), there was a bug where in some cases the index automatically created to provide uniqueness for the existing primary key would have its `EncodingType` field marked as `PrimaryIndexEncoding`. This behavior was fixed in [#66599](https://github.com/cockroachdb/cockroach/pull/66599), but left secondary indexes with primary encoding.
+
+In CockroachDB v22.2 and later, the implementation for [`ALTER TABLE`](../v22.2/alter-table.html#add-column) began using the new declarative schema changer framework. When adding a new column using the new [declarative schema changer](../v22.2/online-schema-changes.html#declarative-schema-changer), CockroachDB retrieves a unique secondary index mistakenly, assuming it to be a primary index. As a result, the secondary index is modified to store the new columns. CockroachDB expects there to be only one primary index, but in the case outlined here, there are two, because a secondary index has been incorrectly labeled as a primary one. This will cause the new column data to be present only in the secondary index and could lead to irrecoverable data loss, if the secondary index is deleted by the user.
```

Comment:
Maybe move the "new declarative schema changer" link earlier to the first mention. I'm also not sure if it's correct to link to the v22.x docs, even though that is when the new one was introduced.

Reply:
I've moved the link here, and changed the version